### PR TITLE
[FEAT] [BWS][BWC]: new endpoint to get nonce

### PR DESCRIPTION
**Pull request by rastajpa**

### CODE REVIEW

**Review by matiu — APPROVED** [review]

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 1]
```diff
@@ -2105,6 +2105,20 @@ export class WalletService {
     });
   }
 
+  getNonce(opts) {
+    const bc = this._getBlockchainExplorer(opts.coin, opts.network);
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/server.ts:2109]: maybe add check that network = eth?

File: packages/bitcore-wallet-service/src/lib/expressapp.ts [hunk 2]
```diff
@@ -1129,6 +1129,28 @@ export class ExpressApp {
       });
     });
 
+    router.get('/v1/nonce/:address', (req, res) => {
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/expressapp.ts:1132]: This request will take a bit, and it is not cached and it is open to everyone. I suggest we do getAuthServer, so at least it has to be attached to a wallet.

**Review by matiu — CHANGES_REQUESTED** [review]
see comments.

File: packages/bitcore-wallet-service/src/lib/expressapp.ts (hunk 2 above)
Comment (rastajpa) [reply packages/bitcore-wallet-service/src/lib/expressapp.ts:1134]: good point. It is definitely better to use getAuthServer.

**Review by gabrielbazan7 — APPROVED** [review]

**Pull request merged**